witness_beacon.go

```diff
@@ -63,6 +73,14 @@ func (p *preimageBeacon) SubscribeUpdates() *contractcourt.WitnessSubscription {
 	}
 }
 
+// castExtpreimageInvoice converts a standard Invoice into one that
+// can be used by the extpreimage package. We do this in a separate function
+// so that we can more easily stub it for testing.
+var castExtpreimageInvoice = func(i channeldb.Invoice) extpreimage.Invoice {
```

Comment:
I know this is weird, it's the best thing I've seen for stubbing something like this.